src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -502,13 +504,26 @@ ReportAie::getPropertyTree20202(const xrt_core::device * _pDevice,
   _pt.add_child("aie_metadata", populate_aie(_pDevice, "Aie_Metadata"));
 }
 
-void 
+void
 ReportAie::writeReport(const xrt_core::device* /*_pDevice*/,
-                       const boost::property_tree::ptree& _pt, 
-                       const std::vector<std::string>& /*_elementsFilter*/, 
+                       const boost::property_tree::ptree& _pt,
+                       const std::vector<std::string>& _elementsFilter,
                        std::ostream & _output) const
 {
   boost::property_tree::ptree empty_ptree;
+  std::vector<std::string> aieCoreList;
+  bool is_less = false;
+
+  for (auto it = _elementsFilter.begin(); it != _elementsFilter.end(); ++it) {
+    if(*it == "cores") {
+      auto core_list = std::next(it);
+      if (core_list != _elementsFilter.end())
+        boost::split(aieCoreList, *core_list, boost::is_any_of(","));
+    }
+    if(*it == "less") {
+      is_less = true;

```

Comment:
Add a comment or rename to something more explicit?
I have just understood what it means by reading your super cool PR description.
The I realize that all this information from the PR description will be lost to the people reading your interesting code.
You could summarize what it does in a big design comment with motivation and example above the display function.

Reply:
Added comments. Please check and share your feedback.